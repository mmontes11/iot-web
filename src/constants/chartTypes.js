export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
