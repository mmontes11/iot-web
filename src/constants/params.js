export const FIRST_PARAM = "FIRST_PARAM";
export const SECOND_PARAM = "SECOND_PARAM";
