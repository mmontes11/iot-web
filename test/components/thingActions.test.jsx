import React from "react";
import { shallow } from "enzyme";
import ThingActions from "components/thingActions";
import { thing } from "../constants";

describe("components/thingActions", () => {
  const onEventStatsClick = jest.fn();
  const onMeasurementStatsClick = jest.fn();
  const wrapper = shallow(
    <ThingActions
      thing={thing}
      onEventStatsClick={onEventStatsClick}
      onMeasurementStatsClick={onMeasurementStatsClick}
    />,
  );
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates clicks on actions", () => {
    wrapper.find("#google-maps-link").simulate("click");
    wrapper.find("#event-stats-button").simulate("click");
    wrapper.find("#measurement-stats-button").simulate("click");
    expect(onEventStatsClick).toHaveBeenCalled();
    expect(onMeasurementStatsClick).toHaveBeenCalled();
  });
});
