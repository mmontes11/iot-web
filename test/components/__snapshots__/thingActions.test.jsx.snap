// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thingActions renders 1`] = `
<p
  className="buttons is-horizontal-center"
>
  <a
    className="button is-medium"
    href="https://www.google.es/maps"
    id="google-maps-link"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="icon is-medium"
    >
      <i
        className="fas fa-map-marked"
      />
    </span>
  </a>
  <button
    className="button is-medium"
    id="stats-button"
    onClick={[Function]}
  >
    <span
      className="icon is-medium"
    >
      <i
        className="fas fa-chart-bar"
      />
    </span>
  </button>
</p>
`;

exports[`components/thingActions renders without actions 1`] = `
<p
  className="buttons is-horizontal-center"
/>
`;
